src/interfaces/validator.js

```diff
@@ -29,16 +30,43 @@ export default async function () {
 		const unique = headers.unique || undefined;
 
 		logger.log('debug', 'Unserializing record');
-		const record = ConversionService.unserialize(data, format);
+		let record = ConversionService.unserialize(data, format);
 
-		if (operation === OPERATIONS.UPDATE && id) {
-			logger.log('debug', `Reading record ${id} from SRU`);
-			const existingRecord = await getRecord(id);
-			logger.log('debug', 'Checking LOW-tag authorization');
-			await OwnAuthorization.validateChanges(cataloger.authorization, record, existingRecord);
-			logger.log('debug', 'Checking CAT field history');
-			validateRecordState(record, existingRecord);
-		} else {
+		logger.log('debug', 'Validating the record');
+		const validationResults = await executeValidations();
+
+		if (noop) {
+			return validationResults;
+		}
+
+		return {headers: {operation, cataloger: cataloger.id}, data: record.toObject()};
+
+		async function executeValidations() {
+			if (operation === OPERATIONS.UPDATE) {
+				return updateValidations();
+			}
+
+			return createValidations();
+		}
+
+		async function updateValidations() {
+			if (id) {
+				record = updateField001ToParamId(`${id}`, record);
```

Comment:
Don't alter the variable but create a new variable instead (GC will handle dangling variables when they aren't used anymore).